.gitignore

```diff
@@ -1,9 +1,18 @@
+# General ignore patterns
 *
```

Comment:
I know this is not related to your PR, but I would remove this line.

Reply:
@jbrzensk, or @jcorbino do you know why `*` was put here in the `.gitignore`? I don't think this is a best practice. We should remove so that would eliminate the need for the exception for the Make file below.
`!Makefile`

Reply:
You can always add the Makefile with git add -f Makefile.

It is a user editable file, which is edited by everyone who downloads it. You have to edit it to build the library. It seems easier to have Git not track your personal changes to the file, by having it in the gitignore.

Same for the doxygen file.

Reply:
Thank you!